### PR TITLE
rock: override built-in checks

### PR DESCRIPTION
**Pull request by DifferentialOrange**
The ability to override built-in modules was introduced with [1]. After [1], checks is embedded to the core Tarantool. The ability to override them with installed rock will make it possible to use old Tarantool with new checks. Core Tarantool tests for master should be fine after this patch, see [3].

1. https://github.com/tarantool/tarantool/issues/7774
2. https://github.com/tarantool/tarantool/issues/7726
3. https://github.com/tarantool/tarantool/pull/8475

### CODE REVIEW

File: rpm/tarantool-checks.spec [hunk 1]
```diff
@@ -22,10 +22,13 @@ Easy, terse, readable and fast function arguments type checking.
 mkdir -p %{br_luapkgdir}
```
Comment (Totktonada) [inline rpm/tarantool-checks.spec:22]: Shouldn't we do `mkdir -p %{br_luapkgdir}/override` as well?

File: rpm/tarantool-checks.spec [hunk 2]
```diff
@@ -22,10 +22,13 @@ Easy, terse, readable and fast function arguments type checking.
...
 cp -av checks.lua %{br_luapkgdir}
 cp -rv checks %{br_luapkgdir}
+cp -av checks.lua %{br_luapkgdir}/override
+cp -rv checks %{br_luapkgdir}/override
 
 %files
 %{luapkgdir}/checks.lua
 %{luapkgdir}/checks
+%{luapkgdir}/override
```
Comment (Totktonada) [inline rpm/tarantool-checks.spec:31]: How about the following?

```
%{luapkgdir}/override/checks.lua
%{luapkgdir}/override/checks
```

I don't know, to be honest, whether it makes any difference, but it seems logical to advertize ownership about files/directories we actually own (where nobody else installs anything).

**Review by Totktonada — APPROVED** [review]

File: rpm/tarantool-checks.spec (hunk 1 above)
Comment (DifferentialOrange) [reply rpm/tarantool-checks.spec:22]: Added

File: rpm/tarantool-checks.spec (hunk 2 above)
Comment (DifferentialOrange) [reply rpm/tarantool-checks.spec:31]: Reworked

**Review by yngvar-antonsson — APPROVED** [review]

**Pull request merged**